pegasus/sites.v3/code.org/views/video_player.js.haml

```diff
@@ -0,0 +1,75 @@
+%script{src: "/js/videojs-ie8.min.js"}
+%script{src: "/js/video.min.js"}
+
+%link{href: "/css/video-js.css", rel: "stylesheet"}
+
+:javascript
+
+  $(window).load(function() {
+    if (window.location.search.indexOf("fallback_video") !== -1)
+    {
+      setupVideos(false);
+    }
+    else
+    {
+      testImageAccess(
+        "https://www.youtube.com/favicon.ico?" + Math.random(),
+        function() { setupVideos("youtube"); },
+        function() { setupVideos("fallback"); });
+    }
+  });
+
+  function setupVideos(player) {
+
+    $(".insert_video_player").each(function() {
+      var downloadPath = $(this).data("download-path");
+
+      // Use fallback player if that's the preference.
+      // It requires a downloadPath, and it doesn't seem to work on IE8 because
+      // it relies upon a missing addEventListener.
+      if (player == "fallback" && downloadPath && window.addEventListener)
+      {
+        $(this).parent().append(
+          '<video id="'+ 'player1' +'" ' +
+          'style="position:absolute; top: 0; left: 0; width: 100%; height: 100%"' +
+          'width="100%" height="100%" ' +
+          'class="video-js" ' +
+          'controls preload="auto" >' +
+          '  <source src="' + downloadPath + '" type="video/mp4"/>' +
+          '</video>');
+      }
+      // Always default to YouTube player.
+      else
+      {
+        $(this).parent().append(
+          '<iframe style="position:absolute; top: 0; left: 0; width: 100%; height: 100%" src="https://www.youtube.com/embed/' + $(this).data("video-code") + '?iv_load_policy=3&rel=0&autohide=1&showinfo=0&enablejsapi=1" frameborder="0" allowfullscreen=true>');
+      }
+    });
+  }
+
+
+  // Tests whether the browser can access an image URL.
+  // Useful as a workaround for CORS security to test access to an origin.
+  function testImageAccess(url, successCallback, failureCallback, timeoutMs) {
```

Comment:
@islemaster or @wjordan might know if there's an simple way to share this some of this logic between pegasus/dashboard's versions, otherwise we can keep an eye out for an opportunity to dedupe if it becomes possible/we make a change to one version